### PR TITLE
Feature organizer email check

### PR DESCRIPTION
**Pull request by bwan2**
Dependency [PR#14](https://github.com/jian-en/pinocchioNodeServer/pull/14)

### Completed
- Added a list of valid domains for the server to check. 
- Added tests/error handling for invalid domains.
- Updated README

### To run
1. Start a dev env `docker-compose -f docker-dev.yml up`
2. Load tables `npm run create-tables`
3. Load initial data `npm run load-initial-data` (This is the list of validDomains that can be found in `src/dynamodb/csvs/validDomains.csv`)
4. Register with valid and invalid domains.

List of valid domains: 
* andrew.cmu.edu
* cs.cmu.edu
* pallet.com (for tests; maybe extract later for only testData)

### CODE REVIEW

File: src/controllers/accounts.controller.js [hunk 1]
```diff
@@ -65,6 +65,16 @@ exports.register = async (req, res, next) => {
             return res.status(422).json(responseMsg.error(errorMsg.params.EMAIL,
                 errorMsg.messages.EMAIL_REGISTERED));
           }
+          // check if the domain is valid
+          const validDomains = await dynamoDb.getValidDomains();
+          const requestDomain = email.split('@').pop();
+          if (!validDomains.success) {
+            return res.status(500).json(validDomains);
+          } else if (!validDomains.data.some((e) => e.email == requestDomain)) {
```
Comment (miajhaochen) [inline src/controllers/accounts.controller.js:73]: for `some`, you can use `indexOf` for simplicity

File: src/controllers/accounts.controller.js [hunk 2]
```diff
@@ -65,6 +65,16 @@ exports.register = async (req, res, next) => {
...
+            return res.status(422).json(responseMsg.error(errorMsg.params.Email,
```
Comment (miajhaochen) [inline src/controllers/accounts.controller.js]: ...... `errorMsg.params.EMAIL `  (EMAIL should be uppercase letters ) 👊👊👊👊👊

File: src/controllers/accounts.controller.js (hunk 1 above)
Comment (miajhaochen) [reply src/controllers/accounts.controller.js:73]: Or, you can take a look at this package: `https://lodash.com/`

File: src/controllers/accounts.controller.js (hunk 2 above)
Comment (bwan2) [reply src/controllers/accounts.controller.js]: ............... thanks

**Pull request merged**